File: backend/domuwa/models/game_category.py
from enum import StrEnum
from typing import TYPE_CHECKING, Optional

from sqlmodel import Field, Relationship, SQLModel

if TYPE_CHECKING:
    from domuwa.models.game_room import GameRoom


class GameCategoryChoices(StrEnum):
    SFW = "SFW"
    NSFW = "NSFW"
    MIXED = "Mixed"


class GameCategoryBase(SQLModel):
    name: GameCategoryChoices


class GameCategory(SQLModel, table=True):
    __tablename__ = "game_category"

    id: Optional[int] = Field(default=None, primary_key=True)
    name: GameCategoryChoices = Field(index=True, unique=True)

    game_rooms: list["GameRoom"] = Relationship(back_populates="game_category")


class GameCategoryCreate(GameCategoryBase):
    pass


class GameCategoryUpdate(GameCategoryBase):
    pass


class GameCategoryRead(GameCategoryBase):
    id: int

File: frontend/src/lib/api/types/game_category.ts
import { z } from "zod";

export enum GameCategoryChoice {
	SFW = "SFW",
	NSFW = "NSFW",
	MIXED = "MIXED",
}

export const GameCategoryCreateSchema = z.object({
	name: z.nativeEnum(GameCategoryChoice),
});
export type GameCategoryCreate = z.infer<typeof GameCategoryCreateSchema>;

export const GameCategoryUpdateSchema = z.object({
	name: z.nativeEnum(GameCategoryChoice),
});
export type GameCategoryUpdate = z.infer<typeof GameCategoryUpdateSchema>;

export const GameCategorySchema = z.object({
	id: z.number().min(1),
	name: z.nativeEnum(GameCategoryChoice),
});
export type GameCategory = z.infer<typeof GameCategorySchema>;

export const GameCategoriesSchema = z.array(GameCategorySchema);
export type GameCategories = z.infer<typeof GameCategoriesSchema>;
